File: frontend/src/types/community-mutes/select-community-mute-dto.ts
import { SelectProfileDTO } from "../profile/select-profile-dto";

export interface SelectCommunityMuteDTO {
    id: string;
    communityId: string;
    mutedProfileId: string;
    mutedProfile?: SelectProfileDTO;
    mutedById: string;
    mutedBy?: SelectProfileDTO;
    muteReason?: string;
    mutedUntil?: string; // ISO DateTime string
    dateCreated?: string; // ISO DateTime string
    
    hasBlockedRelation?: boolean;
    isMutedProfileRestricted?: boolean;
    isMutedByRestricted?: boolean;
    restrictedMessage?: string | null;
    canUnmute?: boolean;
}

File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/CommunityMute/SelectCommunityMuteDTO.cs
using System;
using platform_core_service.Common.Entities.BaseEntity;
using platform_core_service.Common.Models.DTOs.EntityDTO.Profile;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.CommunityMute
{
    public class SelectCommunityMuteDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;
        public string CommunityId { get; set; } = null!;
        public string MutedProfileId { get; set; } = null!;
        public SelectProfileDTO? MutedProfile { get; set; }
        public string MutedById { get; set; } = null!;
        public SelectProfileDTO? MutedBy { get; set; }
        public string? MuteReason { get; set; }
        public DateTimeOffset? MutedUntil { get; set; }
        public DateTimeOffset? DateCreated { get; set; }
        
        // Privacy / relation fields (similar to Ban list)
        public bool HasBlockedRelation { get; set; }
        public bool IsMutedProfileRestricted { get; set; }
        public bool IsMutedByRestricted { get; set; }
        public string? RestrictedMessage { get; set; }
        public bool CanUnmute { get; set; } = true;
    }
}
